language/stdlib/modules/Diem.move

```diff
@@ -622,16 +715,34 @@ module Diem {
     /// at `preburn_address` does not contain a pending burn request.
     public fun cancel_burn_with_capability<CoinType>(
         preburn_address: address,
+        amount: u64,
         _capability: &BurnCapability<CoinType>
     ): Diem<CoinType> acquires CurrencyInfo, Preburn {
-        // destroy the coin in the preburn area
-        assert(exists<Preburn<CoinType>>(preburn_address), Errors::not_published(EPREBURN));
-        let preburn = borrow_global_mut<Preburn<CoinType>>(preburn_address);
-        let coin = withdraw_all<CoinType>(&mut preburn.to_burn);
+        // destroy the coin in the preburn queue with amount
+        assert(exists<ConcurrentPreburns<CoinType>>(preburn_address), Errors::not_published(E_CONCURRENTPREBURNS));
+        let concurrent_preburns = borrow_global_mut<ConcurrentPreburns<CoinType>>(preburn_address);
+        // Abort if no coin present in preburn queue
+
+        assert(len(concurrent_preburns.inflight_burns.value) == 0, Errors::invalid_state(EPREBURN_EMPTY));
+
+        let number_preburns_inflight = Vector::length(concurrent_preburns.inflight_burns);
+        let preburn_index = 0;
+
+        while (preburn_index < number_preburns_inflight) {
```

Comment:
single function to find the coin given amount and concurrent preburn in flight queue